### PR TITLE
Refactor export all my module generator [SCI-5031]

### PR DESCRIPTION
**Pull request by aignatov-bio**
Jira ticket: [SCI-5031](https://biosistemika.atlassian.net/browse/SCI-5031)

### What was done
Refactor export all my module generator

### CODE REVIEW

**Review by okriuchykhin — COMMENTED** [review]
Didn't we discuss tree-like structure?

File: config/initializers/extends.rb [hunk 1]
```diff
@@ -41,6 +41,25 @@ class Extends
                            my_module_repository: 17,
                            my_module_protocol: 18 }
 
+  EXPORT_ALL_PROJECT_ELEMENTS = {
```
Comment (okriuchykhin) [inline config/initializers/extends.rb:44]: So it is not possible to have here all needed arguments for 'gen_element_content' method?

Comment (aignatov-bio) [reply config/initializers/extends.rb:44]: Elements not consistent. In some cases, we iterate and pass only one object, in other, we pass parent object and associate objects. And special cases for results and assigned repositories.

File: config/initializers/extends.rb [hunk 2]
```diff
@@ -41,6 +41,93 @@ class Extends
                            my_module_repository: 17,
                            my_module_protocol: 18 }
 
+  EXPORT_ALL_PROJECT_ELEMENTS = [
+    {
+      type_of: 'project_header',
+      id_key: 'project_id',
+      children: []
```
Comment (okriuchykhin) [inline config/initializers/extends.rb:48]: Maybe just add  null check to avoid specifying empty elements here?

File: app/models/report.rb [hunk 3]
```diff
@@ -155,31 +107,29 @@ def self.generate_whole_project_report(project, current_user, current_team)
...
-        'sort_order' => sort_order.present? ? sort_order : nil,
-        'children' => children.present? ? children : []
+    children.each do |element|
+      element_hash = lambda { |object|
+        hash_object = {
+          'type_of' => element[:type_of] || element[:type_of_lambda].call(object),
+          'id' => { element[:id_key] => object.id },
+          'sort_order' => element[:sort_order],
+          'children' => gen_element_content(object, element[:children])
+        }
```
Comment (okriuchykhin) [inline app/models/report.rb:120]: element_hash['children'] = gen_element_content(object, element[:children]) if element[:children].present?

**Review by okriuchykhin — APPROVED** [review]

**Pull request merged**